app/services/integrations/twitter.ts

```diff
@@ -0,0 +1,156 @@
+import Vue from 'vue';
+import URI from 'urijs';
+import { PersistentStatefulService } from 'services/core/persistent-stateful-service';
+import { Inject } from 'services/core/injector';
+import { handleResponse, authorizedHeaders } from 'util/requests';
+import { mutation } from 'services/core/stateful-service';
+import { Service } from 'services/core';
+import electron from 'electron';
+import { HostsService } from 'services/hosts';
+import Util from 'services/utils';
+import { UserService } from 'services/user';
+import { $t, I18nService } from 'services/i18n';
+import uuid from 'uuid/v4';
+
+interface ITwitterServiceState {
+  linked: boolean;
+  prime: boolean;
+  creatorSiteOnboardingComplete: boolean;
+  creatorSiteUrl: string;
+  screenName: string;
+  tweetWhenGoingLive: boolean;
+}
+
+interface ITwitterStatusResponse {
+  linked: boolean;
+  prime: boolean;
+  cs_onboarding_complete: boolean;
+  cs_url: string;
+  screen_name: string;
+}
+
+export class TwitterService extends PersistentStatefulService<ITwitterServiceState> {
+  @Inject() private hostsService: HostsService;
+  @Inject() private userService: UserService;
+
+  apiToken = this.userService.state.auth.apiToken;
+
+  static defaultState: ITwitterServiceState = {
+    linked: false,
+    prime: false,
+    creatorSiteOnboardingComplete: false,
+    creatorSiteUrl: '',
+    screenName: '',
+    tweetWhenGoingLive: true,
+  };
+
+  @mutation()
+  SET_TWITTER_STATUS(status: ITwitterStatusResponse) {
+    this.state.linked = status.linked;
+    this.state.prime = status.prime;
+    this.state.creatorSiteOnboardingComplete = status.cs_onboarding_complete;
+    this.state.creatorSiteUrl = status.cs_url;
+    this.state.screenName = status.screen_name;
+  }
+
+  @mutation()
+  SET_TWEET_PREFERENCE(preference: boolean) {
+    this.state.tweetWhenGoingLive = preference;
+  }
+
+  setTweetPreference(preference: boolean) {
+    this.SET_TWEET_PREFERENCE(preference);
+  }
+
+  private linkTwitterUrl() {
+    const host = Util.isPreview() ? this.hostsService.beta3 : this.hostsService.streamlabs;
+    const token = this.apiToken;
+    const i18nService = I18nService.instance as I18nService; // TODO: replace with getResource('I18nService')
+    const locale = i18nService.state.locale;
+
+    return `https://${host}/slobs/twitter/link?oauth_token=${token}&l=${locale}`;
```

Comment:
Why we need to use `authorizedHeaders(this.apiToken)` in the `unlinkTwitter` method, but use the GET parameter here?

Reply:
Look at the implementation for dashboard, app store, themes. We are opening auth window on streamlabs.com which is redirected to twitter.com and back. Current implementation of linking twitter gets user info from login session not from a request header.

Reply:
But I did discuss this approach for reducing number of logins required with Salman, Andy, Lohith and we agree that there should be a better way to handle creating authenticated webviews. I will explore the options in the future